File: gramarogya-backend/src/main/java/com/gramarogya/gramarogya_backend/service/NotificationService.java
package com.gramarogya.gramarogya_backend.service;

import com.gramarogya.gramarogya_backend.dto.NotificationResponseDto;

import java.util.List;

public interface NotificationService {

    // ==========================
    // GET ALL NOTIFICATIONS
    // ==========================
    List<NotificationResponseDto> getMyNotifications(String email);

    // ==========================
    // GET LATEST 10 NOTIFICATIONS
    // ==========================
    List<NotificationResponseDto> getLatestNotifications(String email);

    // ==========================
    // GET UNREAD NOTIFICATIONS
    // ==========================
    List<NotificationResponseDto> getUnreadNotifications(String email);

    // ==========================
    // GET UNREAD COUNT
    // ==========================
    long getUnreadCount(String email);

    // ==========================
    // MARK SINGLE AS READ
    // ==========================
    NotificationResponseDto markAsRead(String notificationId, String email);

    // ==========================
    // MARK ALL AS READ
    // ==========================
    void markAllAsRead(String email);

    // ==========================
    // DELETE ONE NOTIFICATION
    // ==========================
    void deleteNotification(String notificationId, String email);

    // ==========================
    // DELETE ALL NOTIFICATIONS
    // ==========================
    void deleteAllNotifications(String email);
}

File: gramarogya_frontend/src/services/notificationService.js
import axiosClient from "../api/axiosClient";

// =======================
// GET ALL NOTIFICATIONS
// =======================
const getNotifications = async () => {
  const response = await axiosClient.get("/notifications");
  return response.data;
};

// =======================
// GET UNREAD NOTIFICATIONS
// =======================
const getUnreadNotifications = async () => {
  const response = await axiosClient.get("/notifications/unread");
  return response.data;
};

// =======================
// GET UNREAD COUNT
// =======================
const getUnreadCount = async () => {
  const response = await axiosClient.get("/notifications/unread/count");
  return response.data;
};

// =======================
// MARK NOTIFICATION AS READ
// =======================
const markAsRead = async (id) => {
  const response = await axiosClient.put(`/notifications/${id}/read`);
  return response.data;
};

// =======================
// MARK ALL AS READ
// =======================
const markAllAsRead = async () => {
  const response = await axiosClient.put("/notifications/read-all");
  return response.data;
};

const notificationService = {
  getNotifications,
  getUnreadNotifications,
  getUnreadCount,
  markAsRead,
  markAllAsRead,
};

export default notificationService;